Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskData/DataStream/DataSource/AbstractEntityProxyInstanceIDDataSource.cs

```diff
@@ -0,0 +1,138 @@
+            
+            if (destination == null)
+            {
+                dependsOn = JobHandle.CombineDependencies(
+                    dependsOn,
+                    PendingData.AcquireAsync(AccessType.ExclusiveWrite));
+            }
+            else
+            {
+                dependsOn = JobHandle.CombineDependencies(
+                    dependsOn,
+                    PendingData.AcquireAsync(AccessType.ExclusiveWrite),
+                    destination.PendingData.AcquireAsync(AccessType.ExclusiveWrite));
+
+                destinationWriter = destination.PendingWriter;
+            }
+            
+            
+            MigrateJob migrateJob = new MigrateJob(
+                PendingData.Pending,
+                destinationWriter,
+                remapArray,
+                destinationWorldDataMap.TaskSetOwnerIDMapping,
+                destinationWorldDataMap.ActiveIDMapping);
+            dependsOn = migrateJob.Schedule(dependsOn);
+
+            PendingData.ReleaseAsync(dependsOn);
+            destination?.PendingData.ReleaseAsync(dependsOn);
+
+            return dependsOn;
+        }
+        
+        [BurstCompile]
+        private struct MigrateJob : IJob
+        {
+            private const int UNSET_ID = -1;
+
+            private UnsafeTypedStream<EntityProxyInstanceID> m_CurrentStream;
+            private readonly UnsafeTypedStream<EntityProxyInstanceID>.Writer m_DestinationStreamWriter;
+            [ReadOnly] private NativeArray<EntityRemapUtility.EntityRemapInfo> m_RemapArray;
+            [ReadOnly] private readonly NativeParallelHashMap<uint, uint> m_TaskSetOwnerIDMapping;
+            [ReadOnly] private readonly NativeParallelHashMap<uint, uint> m_ActiveIDMapping;
+            [NativeSetThreadIndex] private readonly int m_NativeThreadIndex;
+
+            public MigrateJob(
+                UnsafeTypedStream<EntityProxyInstanceID> currentStream,
+                UnsafeTypedStream<EntityProxyInstanceID>.Writer destinationStreamWriter,
+                NativeArray<EntityRemapUtility.EntityRemapInfo> remapArray,
+                NativeParallelHashMap<uint, uint> taskSetOwnerIDMapping,
+                NativeParallelHashMap<uint, uint> activeIDMapping)
+            {
+                m_CurrentStream = currentStream;
+                m_DestinationStreamWriter = destinationStreamWriter;
+                m_RemapArray = remapArray;
+                m_TaskSetOwnerIDMapping = taskSetOwnerIDMapping;
+                m_ActiveIDMapping = activeIDMapping;
+
+                m_NativeThreadIndex = UNSET_ID;
+            }
+
+            public void Execute()
+            {
+                //Can't modify while iterating so we collapse down to a single array and clean the underlying stream.
+                //We'll build this stream back up if anything should still remain
+                NativeArray<EntityProxyInstanceID> currentInstanceArray = m_CurrentStream.ToNativeArray(Allocator.Temp);
```

Comment:
Ohhhh I see what you mean. Interesting. I've added a TODO and referred to this.

I don't think order matters in the stream... removeSwapBack is really neat though, I think that could be possible and is probably the way to do this instead of adding a second stream and double buffering them.

Reply:
yea just want to be explicit in our docs that DataStreams are unordered.